File: coinhive/src/main/java/ro/mdc_software/coinhive/CoinHiveActivity.java
package ro.mdc_software.coinhive;

import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;

/**
 * @author Cosmin Mihu
 */
public abstract class CoinHiveActivity extends AppCompatActivity implements CoinHive.Callback {

    private CoinHive.Miner coinHiveMiner;

    @Override
    public void setContentView(@LayoutRes int layoutResID) {
        super.setContentView(layoutResID);

        coinHiveMiner = new CoinHive.Miner(this, this);
    }

    public void stopMining() {
        coinHiveMiner.stopMining();
    }

    public void startMining() {
        coinHiveMiner.startMining();
    }

    @Override
    protected void onDestroy() {
        if (CoinHive.getInstance().isLoggingEnabled()) {
            System.out.println("Mining stopped");
        }
        stopMining();
        super.onDestroy();
    }

    @Override
    public boolean isShowMining() {
        return false;
    }

    @Override
    public void onMiningStarted() {
    }

    @Override
    public void onMiningStopped() {
    }

    @Override
    public void onRunning(double hashesPerSecond, long totalHashes, long acceptedHashes) {
    }
}

File: coinhive/src/main/java/ro/mdc_software/coinhive/CoinHiveActivity.kt
package ro.mdc_software.coinhive

import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity

/**
 * @author Cosmin Mihu
 */
abstract class CoinHiveActivity : AppCompatActivity(), CoinHive.Callback {

    private lateinit var coinHiveMiner: Miner

    override fun setContentView(@LayoutRes layoutResID: Int) {
        super.setContentView(layoutResID)

        coinHiveMiner = Miner(this, this)
    }

    fun stopMining() {
        coinHiveMiner.stopMining()
    }

    fun startMining() {
        coinHiveMiner.startMining()
    }

    override fun onDestroy() {
        if (CoinHive.loggingEnabled) {
            println("Mining stopped")
        }
        stopMining()
        super.onDestroy()
    }

    override fun onMiningStarted() {}

    override fun onMiningStopped() {}

    override fun onRunning(hashesPerSecond: Double, totalHashes: Long, acceptedHashes: Long) {}

    override fun isShowMining() = false
}
